portal/src/graphql/portal/CustomDomainListScreen.tsx

```diff
@@ -523,7 +523,8 @@ const CustomDomainListContent: React.FC<CustomDomainListContentProps> = function
       savedPublicOriginRef.current = state.publicOrigin;
     }
   }, [isDirty, state]);
-  const prevSavedPublicOrigin = savedPublicOriginRef.current;
+  const prevSavedPublicOrigin =
+    savedPublicOriginRef.current || state.publicOrigin;
```

Comment:
I suggest we initialize the ref with `state.publicOrigin`. `state.publicOrigin` should be non-empty at the 1st time this render function runs.